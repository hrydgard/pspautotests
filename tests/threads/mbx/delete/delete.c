#include "../sub_shared.h"

SETUP_SCHED_TEST;

#define DELETE_TEST(title, mbx) { \
	int result = sceKernelDeleteMbx(mbx); \
	if (result == 0) { \
		printf("%s: OK\n", title); \
	} else { \
		printf("%s: Failed (%X)\n", title, result); \
	} \
}

int main(int argc, char **argv) {
	SceUID mbx = sceKernelCreateMbx("delete1", 0, NULL);

	DELETE_TEST("Normal", mbx);
	DELETE_TEST("NULL", 0);
	DELETE_TEST("Invalid", 0xDEADBEEF);
	DELETE_TEST("Deleted", mbx);

	BASIC_SCHED_TEST("Delete other",
		result = sceKernelDeleteMbx(mbx2);
	);
	BASIC_SCHED_TEST("Delete same",
		result = sceKernelDeleteMbx(mbx1);
	);
	BASIC_SCHED_TEST("NULL",
		result = sceKernelDeleteMbx(0);
	);

	return 0;
}