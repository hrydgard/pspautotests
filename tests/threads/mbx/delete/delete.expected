Normal: OK
NULL: Failed (8002019B)
Invalid: Failed (8002019B)
Deleted: Failed (8002019B)
Delete other: A1B1C2E1D1F2 (thread=800201B5, main=00000000)
Delete same: A1B1C2D1E2F1 (thread=800201B5, main=00000000)
NULL: A1B1C2E1D1F2 (thread=800201B5, main=8002019B)
